manager/dispatcher/attachment.go

```diff
@@ -0,0 +1,75 @@
+package dispatcher
+
+import (
+	"github.com/docker/swarmkit/api"
+	//"github.com/docker/swarmkit/ca"
+	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/state/store"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// CreateNetworkAttachment allows the node to request the resources
+// allocation needed for a network attachment on the specific node.
+// - Returns `InvalidArgument` if the Spec is malformed.
+// - Returns `NotFound` if the Network is not found.
+// - Returns an error if the creation fails.
+func (d *Dispatcher) CreateNetworkAttachment(ctx context.Context, request *api.CreateNetworkAttachmentRequest) (*api.CreateNetworkAttachmentResponse, error) {
```

Comment:
That is session-based rate limiting. I am not sure it generically applies.

What issue with rate limiting do you see?